src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/idconverter/CustomBackendIdConverter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.persistence.idconverter;
+
+import org.springframework.core.convert.converter.Converter;
+import uk.ac.ebi.ampt2d.metadata.persistence.entities.BaseAccessionVersionEntityId;
+
+public class CustomBackendIdConverter implements Converter<String, BaseAccessionVersionEntityId> {
+
+    @Override
+    public BaseAccessionVersionEntityId convert(String id) {
+        if (id.split("\\.").length != 2)
+            throw new IllegalArgumentException("Please provide id in the pattern accession.version");
+        return new BaseAccessionVersionEntityId(id.split("\\.")[0], Integer.parseInt(id.split("\\.")[1]));
```

Comment:
This method splits the same string 3 times. Do it just once and save it in a variable.